Can not change storage directory to external SD card

### Actual behaviour
- the apps stops, if the user attempts to change storage path to external SD

### Expected behaviour
- change storage path to external SD
 
### Steps to reproduce
1. Storage path in preferences is shown as "unknown"
2. click on "Storage path", two directories are shown now
  a) /storage/emulated/0/Android/0/data/com.nextcloud.client/files
  b) /storage/756C-1905/Android/0/data/com.nextcloud.client/files
3. click on the second directory (external SD card)

### Environment data
Android version: 7.1

Device model: Sony Xperia Z

Stock or customized system: customized (Lineage OS 14.1-20170110-UNOFFICIAL-yuga)

Nextcloud app version: 14.1

Nextcloud server version: 11.0.1

### Logs
#### Web server error log
```
192.168.1.15 - sony [28/Jan/2017:21:44:48 +0100] "GET /nextcloud/ocs/v1.php/cloud/users/sony?format=json HTTP/1.1" 200 232 "-" "Mozilla/5.0 (Android) ownCloud-android/1.4.1"
192.168.1.15 - sony [28/Jan/2017:21:44:49 +0100] "GET /nextcloud/index.php/avatar/sony/384 HTTP/1.1" 200 31 "-" "Mozilla/5.0 (Android) ownCloud-android/1.4.1"
```

#### Nextcloud log (data/nextcloud.log)
```
(There is no nextcloud.log? Logcat log is available)
```

@przybylski could you tell me, how to activate logging into nextcloud.log?

I only have the Logcat logs. During the described steps (last step: click on the second directory (external SD card)) I see three exceptions:

```
01-28 21:35:31.876  3296  4269 D OwnCloudClient #15: REQUEST PROPFIND /nextcloud/remote.php/webdav/
01-28 21:35:31.902  2373  2387 I ActivityManager: Waited long enough for: ServiceRecord{e994d3b u0 com.google.android.gms/.stats.service.DropBoxEntryAddedService}
01-28 21:35:32.175 26167  4080 W PlatformStatsUtil: Could not retrieve Usage & Diagnostics setting. Giving up.
01-28 21:35:32.254  3296  4268 D GetRemoteSharesForFileOperation: Got 0 shares
01-28 21:35:32.309  3296  4268 D FileDataStorageManager: Sending 5 operations to FileContentProvider
01-28 21:35:32.309  3296  4268 D FileContentProvider: applying batch in provider com.owncloud.android.providers.FileContentProvider@789c66d (temporary: false)
01-28 21:35:32.355  3296  4268 D FileContentProvider: applied batch in provider com.owncloud.android.providers.FileContentProvider@789c66d
01-28 21:35:32.355  3296  4268 D RefreshFolderOperation: Send broadcast com.owncloud.android.operations.RefreshFolderOperation.EVENT_SINGLE_FOLDER_SHARES_SYNCED
01-28 21:35:32.358  3296  3296 D FileDisplayActivity: Received broadcast com.owncloud.android.operations.RefreshFolderOperation.EVENT_SINGLE_FOLDER_SHARES_SYNCED
01-28 21:35:32.395  3296  3296 D FileDisplayActivity: Setting progress visibility to false
01-28 21:35:32.419  3296  4257 D SimpleFactoryManager: getClientFor(OwnCloudAccount ... : 
01-28 21:35:32.420  3296  4257 D OwnCloudClient #22: Creating OwnCloudClient
01-28 21:35:32.423  3296  4257 V SimpleFactoryManager:     new client {sony@192.168.1.2/nextcloud, 26777861}
01-28 21:35:32.434  3296  4257 D Thumbnail: URI: http://192.168.1.2/nextcloud/index.php/apps/files/api/v1/thumbnail/384/384/Nextcloud.mp4
01-28 21:35:32.435  3296  4257 D OwnCloudClient #22: REQUEST GET /nextcloud/index.php/apps/files/api/v1/thumbnail/384/384/Nextcloud.mp4
01-28 21:35:32.436  3296  4269 D WebdavEntry: QUOTA_USED_BYTES 33161766
01-28 21:35:32.437  3296  4269 D WebdavEntry: QUOTA_AVAILABLE_BYTES -3
01-28 21:35:32.439  3296  4269 D WebdavEntry: QUOTA_USED_BYTES 78496
01-28 21:35:32.439  3296  4269 D WebdavEntry: QUOTA_AVAILABLE_BYTES -3
01-28 21:35:32.445  3296  4269 D WebdavEntry: QUOTA_USED_BYTES 2360011
01-28 21:35:32.445  3296  4269 D WebdavEntry: QUOTA_AVAILABLE_BYTES -3
01-28 21:35:32.448  3296  4269 D WebdavEntry: QUOTA_USED_BYTES 26087031
01-28 21:35:32.448  3296  4269 D WebdavEntry: QUOTA_AVAILABLE_BYTES -3
01-28 21:35:32.449  3296  4269 I ReadRemoteFolderOperation: Synchronized /: Operation finished with HTTP status code 207 (success)
01-28 21:35:32.449  3296  4269 D RefreshFolderOperation: Synchronizing sony@192.168.1.2/nextcloud/
01-28 21:35:32.454  3296  4269 D RefreshFolderOperation: Remote folder / changed - starting update of local data 
01-28 21:35:32.462  3296  4269 D FileDataStorageManager: Saving folder / with 5 children and 0 files to remove
01-28 21:35:32.488  3296  4269 D FileDataStorageManager: Sending 6 operations to FileContentProvider
01-28 21:35:32.488  3296  4269 D FileContentProvider: applying batch in provider com.owncloud.android.providers.FileContentProvider@789c66d (temporary: false)
01-28 21:35:32.596  3296  4269 D FileContentProvider: applied batch in provider com.owncloud.android.providers.FileContentProvider@789c66d
01-28 21:35:32.597  3296  4269 D RefreshFolderOperation: Send broadcast com.owncloud.android.operations.RefreshFolderOperation.EVENT_SINGLE_FOLDER_CONTENTS_SYNCED
01-28 21:35:32.639  3296  3296 D FileDisplayActivity: Received broadcast com.owncloud.android.operations.RefreshFolderOperation.EVENT_SINGLE_FOLDER_CONTENTS_SYNCED
01-28 21:35:32.642  3296  4269 D OwnCloudClient #15: REQUEST GET /nextcloud/ocs/v1.php/apps/files_sharing/api/v1/shares
01-28 21:35:32.680  3296  3296 D FileDisplayActivity: Setting progress visibility to true
01-28 21:35:33.203 26167 26218 I Icing   : Usage reports 0 indexed 0 rejected 0 imm upload false
01-28 21:35:33.222  3296  4269 D GetRemoteSharesForFileOperation: Got 0 shares
01-28 21:35:33.263  3296  4269 D FileDataStorageManager: Sending 5 operations to FileContentProvider
01-28 21:35:33.263  3296  4269 D FileContentProvider: applying batch in provider com.owncloud.android.providers.FileContentProvider@789c66d (temporary: false)
01-28 21:35:33.295  3296  4269 D FileContentProvider: applied batch in provider com.owncloud.android.providers.FileContentProvider@789c66d
01-28 21:35:33.296  3296  4269 D RefreshFolderOperation: Send broadcast com.owncloud.android.operations.RefreshFolderOperation.EVENT_SINGLE_FOLDER_SHARES_SYNCED
01-28 21:35:33.303  3296  3296 D FileDisplayActivity: Received broadcast com.owncloud.android.operations.RefreshFolderOperation.EVENT_SINGLE_FOLDER_SHARES_SYNCED
01-28 21:35:33.336  3296  3296 D FileDisplayActivity: Setting progress visibility to false
01-28 21:35:33.383  3296  3311 I art     : Do full code cache collection, code=96KB, data=126KB
01-28 21:35:33.384  3296  3311 I art     : Starting a blocking GC JitCodeCache
01-28 21:35:33.384  3296  3311 I art     : After code cache collection, code=90KB, data=99KB
01-28 21:35:34.881  2074  2157 D audio_hw_primary: out_set_parameters: enter: usecase(1: low-latency-playback) kvpairs: routing=2
01-28 21:35:34.885  2074  2157 D audio_hw_primary: start_output_stream: enter: stream(0xb61bd100)usecase(1: low-latency-playback) devices(0x2)
01-28 21:35:34.885  2074  2157 D audio_hw_primary: select_devices: out_snd_device(2: speaker) in_snd_device(0: )
01-28 21:35:34.886  2074  2157 I msm8960_platform: platform_send_audio_calibration: sending audio calibration for snd_device(2) acdb_id(15)
01-28 21:35:34.886  2074  2157 D ACDB-LOADER: ACDB -> send_afe_cal
01-28 21:35:34.900  2074  2157 D audio_hw_primary: select_devices: done
01-28 21:35:36.027 26167 26218 I Icing   : Indexing 80BD8CA70FA4D0586BC1331514999ADA7E76FB61 from com.google.android.gms
01-28 21:35:36.321 26167 26218 I Icing   : Usage reports 0 indexed 0 rejected 0 imm upload false
01-28 21:35:36.337 26167 26218 I Icing   : Indexing done 2F982CB8329CBDF506523B6AD7D8180650895DA0
01-28 21:35:36.350 26167  4151 W art     : Long monitor contention with owner [bys] processing com.google.android.gms.icing.proxy.UpdateIcingCorporaIntentOperation for action android.intent.action.LOCALE_CHANGED (4150) at void byv.run()(:com.google.android.gms:1857) waiters=0 in void byv.run() for 17.414s
01-28 21:35:36.388 26167 26218 I Icing   : Indexing done 80BD8CA70FA4D0586BC1331514999ADA7E76FB61
01-28 21:35:36.392 26167 26218 I Icing   : Indexing 635FD6F24B41255A47ED1DF6165D436AFE308654 from com.google.android.gms
01-28 21:35:36.427 26167  4319 W IcingInternalCorpora: getNumBytesRead when not calculated.
01-28 21:35:37.261 26167  4080 W PlatformStatsUtil: Could not retrieve Usage & Diagnostics setting. Giving up.
01-28 21:35:37.309 26167 26218 I Icing   : Indexing done 635FD6F24B41255A47ED1DF6165D436AFE308654
01-28 21:35:37.335 26167 26218 I Icing   : Indexing 4AA78D953181689A49AE87DD679106DDCDD61B72 from com.google.android.gms
01-28 21:35:37.549 26167 26218 I Icing   : Indexing done 4AA78D953181689A49AE87DD679106DDCDD61B72
01-28 21:35:37.865  2074  2157 D audio_hw_primary: out_set_parameters: enter: usecase(1: low-latency-playback) kvpairs: routing=2
01-28 21:35:38.298  2373  3226 I ActivityManager: START u0 {cmp=com.nextcloud.client/com.owncloud.android.ui.activity.Preferences} from uid 10124 on display 0
01-28 21:35:38.365  3296  3296 V FileDisplayActivity: onPause() start
01-28 21:35:38.367  3296  3296 D FileDisplayActivity: onPause() ending
01-28 21:35:38.367  3296  3296 V FileDisplayActivity: onPause() end
01-28 21:35:38.397  3296  3296 D Preferences: onCreate(Bundle) starting
01-28 21:35:38.468  3296  3296 D Preferences: onStart() starting
01-28 21:35:38.469  3296  4341 D SimpleFactoryManager: getClientFor(OwnCloudAccount ... : 
01-28 21:35:38.471  3296  4341 D OwnCloudClient #23: Creating OwnCloudClient
01-28 21:35:38.471  3296  4341 V SimpleFactoryManager:     new client {sony@192.168.1.2/nextcloud, 203081788}
01-28 21:35:38.473  3296  3296 D Preferences: onResume() starting
01-28 21:35:38.707  2373  2394 I ActivityManager: Displayed com.nextcloud.client/com.owncloud.android.ui.activity.Preferences: +338ms
01-28 21:35:39.085  3296  3296 V FileDisplayActivity: onSaveInstanceState() start
01-28 21:35:39.088  3296  3296 D FileDisplayActivity: onSaveInstanceState(Bundle) starting
01-28 21:35:39.088  3296  3296 D ExtendedListFragment: onSaveInstanceState()
01-28 21:35:39.089  3296  3296 V FileDisplayActivity: onSaveInstanceState() end
01-28 21:35:39.097  3296  3296 V FileDisplayActivity: onStop() start
01-28 21:35:39.098  3296  3296 D FileDisplayActivity: onStop() ending
01-28 21:35:39.099  3296  3296 V FileDisplayActivity: onStop() end
01-28 21:35:41.014  2074  2157 D audio_hw_primary: out_standby: enter: stream (0xb61bd100) usecase(1: low-latency-playback)
01-28 21:35:42.656  2074  2157 D audio_hw_primary: out_set_parameters: enter: usecase(1: low-latency-playback) kvpairs: routing=2
01-28 21:35:42.658  2074  2157 D audio_hw_primary: start_output_stream: enter: stream(0xb61bd100)usecase(1: low-latency-playback) devices(0x2)
01-28 21:35:42.659  2074  2157 D audio_hw_primary: select_devices: out_snd_device(2: speaker) in_snd_device(0: )
01-28 21:35:42.659  2074  2157 I msm8960_platform: platform_send_audio_calibration: sending audio calibration for snd_device(2) acdb_id(15)
01-28 21:35:42.659  2074  2157 D ACDB-LOADER: ACDB -> send_afe_cal
01-28 21:35:42.668  2074  2157 D audio_hw_primary: select_devices: done
01-28 21:35:44.997  2074  2157 D audio_hw_primary: out_set_parameters: enter: usecase(1: low-latency-playback) kvpairs: routing=2
01-28 21:35:45.142  2373 14896 W InputMethodManagerService: Window already focused, ignoring focus gain of: com.android.internal.view.IInputMethodClient$Stub$Proxy@359c844 attribute=null, token = android.os.BinderProxy@5114467
01-28 21:35:45.151  3296  4258 E AndroidRuntime: FATAL EXCEPTION: AsyncTask #5
01-28 21:35:45.151  3296  4258 E AndroidRuntime: Process: com.nextcloud.client, PID: 3296
01-28 21:35:45.151  3296  4258 E AndroidRuntime: java.lang.RuntimeException: An error occurred while executing doInBackground()
01-28 21:35:45.151  3296  4258 E AndroidRuntime: 	at android.os.AsyncTask$3.done(AsyncTask.java:325)
01-28 21:35:45.151  3296  4258 E AndroidRuntime: 	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
01-28 21:35:45.151  3296  4258 E AndroidRuntime: 	at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
01-28 21:35:45.151  3296  4258 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:242)
01-28 21:35:45.151  3296  4258 E AndroidRuntime: 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
01-28 21:35:45.151  3296  4258 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
01-28 21:35:45.151  3296  4258 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
01-28 21:35:45.151  3296  4258 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:761)
01-28 21:35:45.151  3296  4258 E AndroidRuntime: Caused by: java.lang.ArrayIndexOutOfBoundsException: length=0; index=0
01-28 21:35:45.151  3296  4258 E AndroidRuntime: 	at com.owncloud.android.ui.activity.StorageMigration$FileMigrationTaskBase.restoreAccountsSyncStatus(StorageMigration.java:246)
01-28 21:35:45.151  3296  4258 E AndroidRuntime: 	at com.owncloud.android.ui.activity.StorageMigration$FileMigrationTask.doInBackground(StorageMigration.java:345)
01-28 21:35:45.151  3296  4258 E AndroidRuntime: 	at com.owncloud.android.ui.activity.StorageMigration$FileMigrationTask.doInBackground(StorageMigration.java:284)
01-28 21:35:45.151  3296  4258 E AndroidRuntime: 	at android.os.AsyncTask$2.call(AsyncTask.java:305)
01-28 21:35:45.151  3296  4258 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
01-28 21:35:45.151  3296  4258 E AndroidRuntime: 	... 4 more
01-28 21:35:45.186  2373  2745 W ActivityManager:   Force finishing activity com.nextcloud.client/com.owncloud.android.ui.activity.Preferences
01-28 21:35:45.229  2373  2387 I ActivityManager: Killing 693:com.google.android.gms.feedback/u0a35 (adj 906): empty #17
01-28 21:35:45.252  3296  3296 D Preferences: onPause() ending
01-28 21:35:45.276  3296  3296 V BaseActivity: onRestart() start
01-28 21:35:45.282  3296  3296 V BaseActivity: onRestart() end
01-28 21:35:45.286  3296  3296 V FileDisplayActivity: onStart() start
01-28 21:35:45.289  3296  3296 D FileDisplayActivity: onStart() starting
01-28 21:35:45.317  3296  3296 D FileDisplayActivity: Initializing Fragments in onAccountChanged..
01-28 21:35:45.325  2373  3373 D GraphicsStats: Buffer count: 11
01-28 21:35:45.327  2373  9941 D ActivityManager: cleanUpApplicationRecord -- 693
01-28 21:35:45.328  3296  4436 D SimpleFactoryManager: getClientFor(OwnCloudAccount ... : 
01-28 21:35:45.330  3296  4436 D OwnCloudClient #24: Creating OwnCloudClient
01-28 21:35:45.332  3296  4436 V SimpleFactoryManager:     new client {sony@192.168.1.2/nextcloud, 158297092}
01-28 21:35:45.352  3296  4270 D SimpleFactoryManager: getClientFor(OwnCloudAccount ... : 
01-28 21:35:45.353  3296  4436 D OwnCloudClient #24: REQUEST GET /nextcloud/ocs/v1.php/cloud/users/sony
01-28 21:35:45.360  3296  4270 D OwnCloudClient #25: Creating OwnCloudClient
01-28 21:35:45.364  3296  4270 V SimpleFactoryManager:     new client {sony@192.168.1.2/nextcloud, 108160493}
01-28 21:35:45.393  3296  3296 V FileDisplayActivity: onStart() end
01-28 21:35:45.395  3296  3296 V FileDisplayActivity: onResume() start
01-28 21:35:45.396  3296  3296 D FileDisplayActivity: onResume() starting
01-28 21:35:45.410  3296  4270 D Avatar  : URI: http://192.168.1.2/nextcloud/index.php/avatar/sony/384
01-28 21:35:45.438  2373  2387 W BroadcastQueue: Skipping deliver [background] BroadcastRecord{46b8a4 u-1 com.owncloud.android.syncadapter.FileSyncAdapter.EVENT_FULL_SYNC_START} to ReceiverList{6cb9036 3296 com.nextcloud.client/10124/u0 remote:32f1dd1}: process crashing
01-28 21:35:45.442  3296  3296 V FileDisplayActivity: onResume() end
01-28 21:35:45.469  2373   729 I OpenGLRenderer: Initialized EGL, version 1.4
01-28 21:35:45.470  2373   729 D OpenGLRenderer: Swap behavior 1
01-28 21:35:45.472  2373  2383 I art     : Background sticky concurrent mark sweep GC freed 153557(8MB) AllocSpace objects, 5(100KB) LOS objects, 23% free, 28MB/37MB, paused 5.126ms total 313.995ms
01-28 21:35:45.495  3296  4270 D OwnCloudClient #25: REQUEST GET /nextcloud/index.php/avatar/sony/384
01-28 21:35:46.023  3296  3296 D Preferences: onStop() ending
01-28 21:35:46.027  3296  3296 D Preferences: onDestroy() ending
01-28 21:35:46.073  3296  3296 E WindowManager: 
01-28 21:35:46.073  3296  3296 E WindowManager: android.view.WindowLeaked: Activity com.owncloud.android.ui.activity.Preferences has leaked window DecorView@e6c092b[Updating storage path] that was originally added here
01-28 21:35:46.073  3296  3296 E WindowManager: 	at android.view.ViewRootImpl.<init>(ViewRootImpl.java:418)
01-28 21:35:46.073  3296  3296 E WindowManager: 	at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:331)
01-28 21:35:46.073  3296  3296 E WindowManager: 	at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:94)
01-28 21:35:46.073  3296  3296 E WindowManager: 	at android.app.Dialog.show(Dialog.java:322)
01-28 21:35:46.073  3296  3296 E WindowManager: 	at com.owncloud.android.ui.activity.StorageMigration.migrate(StorageMigration.java:74)
01-28 21:35:46.073  3296  3296 E WindowManager: 	at com.owncloud.android.ui.activity.Preferences$9.onPreferenceChange(Preferences.java:361)
01-28 21:35:46.073  3296  3296 E WindowManager: 	at android.preference.Preference.callChangeListener(Preference.java:939)
01-28 21:35:46.073  3296  3296 E WindowManager: 	at android.preference.ListPreference.onDialogClosed(ListPreference.java:282)
01-28 21:35:46.073  3296  3296 E WindowManager: 	at android.preference.DialogPreference.onDismiss(DialogPreference.java:402)
01-28 21:35:46.073  3296  3296 E WindowManager: 	at android.app.Dialog$ListenersHandler.handleMessage(Dialog.java:1343)
01-28 21:35:46.073  3296  3296 E WindowManager: 	at android.os.Handler.dispatchMessage(Handler.java:102)
01-28 21:35:46.073  3296  3296 E WindowManager: 	at android.os.Looper.loop(Looper.java:154)
01-28 21:35:46.073  3296  3296 E WindowManager: 	at android.app.ActivityThread.main(ActivityThread.java:6126)
01-28 21:35:46.073  3296  3296 E WindowManager: 	at java.lang.reflect.Method.invoke(Native Method)
01-28 21:35:46.073  3296  3296 E WindowManager: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
01-28 21:35:46.073  3296  3296 E WindowManager: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
01-28 21:35:46.090  3296  4270 D skia    : --- SkAndroidCodec::NewFromStream returned null
01-28 21:35:46.097  3296  4270 D SimpleFactoryManager: getClientFor(OwnCloudAccount ... : 
01-28 21:35:46.099  3296  4270 D OwnCloudClient #26: Creating OwnCloudClient
01-28 21:35:46.104  3296  4270 V SimpleFactoryManager:     new client {sony@192.168.1.2/nextcloud, 249935649}
01-28 21:35:46.127  3296  4270 D Thumbnail: URI: http://192.168.1.2/nextcloud/index.php/apps/files/api/v1/thumbnail/384/384/Nextcloud.mp4
01-28 21:35:46.129  2373  6124 W WindowManager: Failed looking up window
01-28 21:35:46.129  2373  6124 W WindowManager: java.lang.IllegalArgumentException: Requested window android.os.BinderProxy@18beed6 does not exist
01-28 21:35:46.129  2373  6124 W WindowManager: 	at com.android.server.wm.WindowManagerService.windowForClientLocked(WindowManagerService.java:9440)
01-28 21:35:46.129  2373  6124 W WindowManager: 	at com.android.server.wm.WindowManagerService.windowForClientLocked(WindowManagerService.java:9431)
01-28 21:35:46.129  2373  6124 W WindowManager: 	at com.android.server.wm.WindowManagerService.removeWindow(WindowManagerService.java:2394)
01-28 21:35:46.129  2373  6124 W WindowManager: 	at com.android.server.wm.Session.remove(Session.java:193)
01-28 21:35:46.129  2373  6124 W WindowManager: 	at android.view.IWindowSession$Stub.onTransact(IWindowSession.java:242)
01-28 21:35:46.129  2373  6124 W WindowManager: 	at com.android.server.wm.Session.onTransact(Session.java:136)
01-28 21:35:46.129  2373  6124 W WindowManager: 	at android.os.Binder.execTransact(Binder.java:565)
01-28 21:35:46.138  3296  4270 D OwnCloudClient #26: REQUEST GET /nextcloud/index.php/apps/files/api/v1/thumbnail/384/384/Nextcloud.mp4
01-28 21:35:48.141  2074  2157 D audio_hw_primary: out_standby: enter: stream (0xb61bd100) usecase(1: low-latency-playback)
```

--- a/src/com/owncloud/android/ui/activity/StorageMigration.java
+++ b/src/com/owncloud/android/ui/activity/StorageMigration.java
@@ -175,7 +175,7 @@ public class StorageMigration {
                                      String source,
                                      String target,
                                      ProgressDialog progressDialog,
-                                     StorageMigrationProgressListener listener) {
+                                     StorageMigrationProgressListener listener) throws SecurityException {
             mContext = context;
             mStorageSource = source;
             mStorageTarget = target;
@@ -242,6 +242,12 @@ public class StorageMigration {
         }
 
         protected void restoreAccountsSyncStatus(boolean oldSync[]) {
+            // If we don't have the old sync statuses, then
+            // probably migration failed even before saving states,
+            // which is weird and should be investigated.
+            // But its better than crashing on ArrayOutOfBounds.
+            if (oldSync == null)
+                return;
             for (int i = 0; i < mOcAccounts.length; ++i) {
                 ContentResolver.setSyncAutomatically(mOcAccounts[i], mAuthority, oldSync[i]);
             }
@@ -263,7 +269,7 @@ public class StorageMigration {
             publishProgress(R.string.file_migration_preparing);
 
             Log_OC.stopLogging();
-            boolean[] syncStates = new boolean[0];
+            boolean[] syncStates = null;
             try {
                 publishProgress(R.string.file_migration_saving_accounts_configuration);
                 syncStates = saveAccountsSyncStatus();
@@ -306,7 +312,7 @@ public class StorageMigration {
             publishProgress(R.string.file_migration_preparing);
             Log_OC.stopLogging();
 
-            boolean[] syncState = new boolean[0];
+            boolean[] syncState = null;
 
             try {
                 File dstFile = new File(mStorageTarget + File.separator + MainApp.getDataFolder());

